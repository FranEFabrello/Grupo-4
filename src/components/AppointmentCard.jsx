import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';

export default function AppointmentCard({
  day,
  time,
  doctor,
  specialty,
  onCancel,
  onConfirm,
  status,
  bgColor = 'bg-blue-100',
  colorScheme,
  showActions = false, // nuevo prop para mostrar botones
}) {
  const cardBgClass = colorScheme === 'light' ? 'bg-white' : 'bg-gray-800';
  const dateBgClass = colorScheme === 'light' ? bgColor : 'bg-blue-900';
  const dateTextClass = colorScheme === 'light' ? 'text-blue-600' : 'text-blue-300';
  const doctorTextClass = colorScheme === 'light' ? 'text-gray-800' : 'text-gray-200';
  const specialtyTextClass = colorScheme === 'light' ? 'text-gray-600' : 'text-gray-400';
  const cancelButtonClass = colorScheme === 'light' ? 'bg-red-600' : 'bg-red-700';
  //const confirmButtonClass = colorScheme === 'light' ? 'border-green-600' : 'border-green-500';
  //const confirmTextClass = colorScheme === 'light' ? 'text-green-600' : 'text-green-500';
  //const confirmIconColor = colorScheme === 'light' ? '#16a34a' : '#22c55e';

  return (
    <View className={`rounded-lg p-4 shadow-md ${cardBgClass} ${cardBgClass}`}>
      <View className="flex-row items-center h-full">
        <View className={`w-20 ${dateBgClass} rounded-lg p-2 items-center`}>
          <Text className={`text-xs ${dateTextClass}`}>{day}</Text>
          <Text className={`text-sm font-semibold ${dateTextClass}`}>{time}</Text>
        </View>
        <View className="ml-3 flex-1 justify-center">
          <Text className={`text-base font-semibold ${doctorTextClass}`}>{doctor}</Text>
          <Text className={`text-sm ${specialtyTextClass}`}>{specialty}</Text>
          {showActions && onCancel && (
            <TouchableOpacity
              className={`${cancelButtonClass} rounded-lg p-2 flex-row justify-center items-center mt-3`}
              onPress={onCancel}
            >
              <Icon name="times" size={14} color="white" />
              <Text className="text-white text-sm ml-2">Cancelar turno</Text>
            </TouchableOpacity>
          )}
        </View>
      </View>
    </View>
  );
}