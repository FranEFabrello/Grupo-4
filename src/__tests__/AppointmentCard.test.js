import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import AppointmentCard from '~/components/AppointmentCard';

jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon');
jest.mock('react-i18next', () => ({ useTranslation: () => ({ t: key => key, i18n: {} }) }));

describe('AppointmentCard', () => {
  const props = {
    day: '10',
    time: '15:00',
    doctor: 'Dr. House',
    specialty: 'Diagnostics',
    status: 'confirmed',
    onPress: jest.fn(),
    colorScheme: 'light',
  };

  it('renders correctly and matches snapshot', () => {
    const { toJSON, getByText } = render(<AppointmentCard {...props} />);
    expect(getByText('10')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when pressed', () => {
    const { getByText } = render(<AppointmentCard {...props} />);
    fireEvent.press(getByText('10'));
    expect(props.onPress).toHaveBeenCalled();
  });
});

