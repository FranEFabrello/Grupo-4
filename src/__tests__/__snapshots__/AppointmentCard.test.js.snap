// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AppointmentCard renders correctly and matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    className="rounded-lg p-4 shadow-md bg-white flex-row items-center"
  >
    <View
      className="bg-blue-100 rounded-lg items-center justify-center px-3 py-2"
      style={
        {
          "alignSelf": "flex-start",
          "minHeight": 70,
        }
      }
    >
      <Text
        className="text-xs text-blue-600"
      >
        10
      </Text>
      <Text
        className="text-sm font-semibold mt-1 text-blue-600"
      >
        15:00
      </Text>
    </View>
    <View
      className="ml-3 justify-center"
    >
      <Text
        className="text-base font-semibold text-gray-800"
        numberOfLines={1}
      >
        Dr. House
      </Text>
      <Text
        className="text-sm text-gray-600"
      >
        Diagnostics
      </Text>
    </View>
  </View>
</View>
`;
